fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/db2/Db2Adapter.java

```diff
@@ -365,43 +365,83 @@ public void detachPartition(String schemaName, String tableName, String partitio
         final String qname = DataDefinitionUtil.getQualifiedName(schemaName, tableName);
         final String detachedName = DataDefinitionUtil.getQualifiedName(schemaName, intoTableName);
         final String ddl = "ALTER TABLE " + qname + " DETACH PARTITION " + partitionName + " INTO " + detachedName;
-        runStatement(ddl);
+        
+        try {
+            runStatement(ddl);
+        }
+        catch (DataAccessException x) {
+            // Suppress the error, in case this is an older version and we have a new table
+            logger.warning("Detach partition skipped for '" + qname + "/" + partitionName + "'. Reason: " + x.getMessage());
```

Comment:
this does remind me a question, if a new table is added to the schema during the schema upgrade, will the according partitions be created automatically for all tenants in this schema?

Reply:
and if loadPartitionInfoMap(partitionInfoMap, schemaName) get partition info from database partition system table, why can we still run into this dataaccesexeption? I mean if a partition already has already been created in a table, not sure why this can fail ...

Reply:
Good point about adding existing tenant partitions to new tables. I'll create an issue for that. It's fairly easy because all the partitions can be specified in the initial create table DDL.

Regarding loadPartitionInfoMap, we can still run into an exception there if something goes wrong with the database connection, but at that point it's game over, can't continue.

Reply:
creating an issue for partitions creation for existing tenants for new table is great, thanks!